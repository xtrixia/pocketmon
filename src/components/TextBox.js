import { css } from "@emotion/css";
import clsx from "clsx";

import { COLORS } from "../root/colors";
import { SPACINGS } from "../root/spacings";
import { BREAKPOINTS } from "../root/breakpoints";

function TextBox({
  id,
  isError,
  className,
  fullWidth,
  helperText,
  label,
  placeholder,
  value,
  width,
  onSearch,
}) {
  return (
    <fieldset
      className={clsx(
        className,
        css`
          display: flex;
          flex-direction: column;
        `
      )}
    >
      <label
        htmlFor={id}
        className={css`
          padding-left: ${SPACINGS.sm};
          padding-bottom: ${SPACINGS.xxs};
          font-size: 1.25rem;
          @media (min-width: ${BREAKPOINTS.sm}) {
            font-size: 2rem;
          }
        `}
      >
        {label}
      </label>
      <input
        id={id}
        name={id}
        type="text"
        placeholder={placeholder}
        className={css`
          width: ${fullWidth ? "100%" : width ? width : "auto"};
          font-family: Karla, sans-serif;
          border: 1px solid ${COLORS.dark};
          border-radius: 2rem;
          box-shadow: 1px 2px;
          padding: ${SPACINGS.xs} ${SPACINGS.sm};
          font-size: 1.25rem;
          @media (min-width: ${BREAKPOINTS.sm}) {
            font-size: 2rem;
          }
        `}
        value={value}
        onChange={onSearch}
      />
      <span
        className={css`
          padding: ${SPACINGS.xxs} ${SPACINGS.sm};
          color: ${isError ? COLORS.danger : COLORS.others};
          font-size: 1rem;
          @media (min-width: ${BREAKPOINTS.sm}) {
            font-size: 1.25rem;
          }
        `}
      >
        {helperText}
      </span>
    </fieldset>
  );
}
export default TextBox;
