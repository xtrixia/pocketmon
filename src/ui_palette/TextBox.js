import { css } from "@emotion/css";
import { COLORS } from "../root/colors";
import { SPACINGS } from "../root/spacings";

function SearchBar({ id, fullWidth, label, placeholder, value, onSearch }) {
  return (
    <div
      className={css`
        display: flex;
        flex-direction: column;
      `}
    >
      <label
        for={id}
        className={css`
          padding-left: ${SPACINGS.sm};
          padding-bottom: ${SPACINGS.xxs};
        `}
      >
        {label}
      </label>
      <input
        id={id}
        name={id}
        type="text"
        placeholder={placeholder}
        className={css`
          width: ${fullWidth ? "100%" : "fit-content"};
          font-family: Karla, sans-serif;
          border: 1px solid ${COLORS.dark};
          border-radius: 2rem;
          box-shadow: 1px 2px;
          padding: ${SPACINGS.xs} ${SPACINGS.sm};
        `}
        value={value}
        onClick={onSearch}
      />
    </div>
  );
}
export default SearchBar;
