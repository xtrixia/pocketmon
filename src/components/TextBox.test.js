import React from "react";
import renderer from "react-test-renderer";

import TextBox from "./TextBox";

test("TextBox should render correctly according to passed props", () => {
  const component = renderer.create(
    <TextBox
      id="input-nickname-field"
      label="Nickname"
      value=""
      placeholder="Input nickname here"
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.type).toEqual("fieldset");

  const label = tree.children[0];
  expect(label.props).toHaveProperty("htmlFor", "input-nickname-field");
  expect(label.children[0]).toEqual("Nickname");

  const input = tree.children[1];
  expect(input.children).toBeNull();
});

test("TextBox should render with helper text below", () => {
  const component = renderer.create(
    <TextBox
      value=""
      helperText="This is just short info"
      placeholder="Input short info about you"
    />
  );
  
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const span = tree.children[2];
  expect(span.type).toEqual("span");
  expect(span.children[0]).toEqual("This is just short info");
});
