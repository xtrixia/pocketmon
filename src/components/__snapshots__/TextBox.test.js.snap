// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox should render correctly according to passed props 1`] = `
<fieldset
  className="css-ur5llr"
>
  <label
    className="css-1a6u95k"
    htmlFor="input-nickname-field"
  >
    Nickname
  </label>
  <input
    className="css-1cn7waq"
    id="input-nickname-field"
    name="input-nickname-field"
    placeholder="Input nickname here"
    type="text"
    value=""
  />
  <span
    className="css-1eztcs5"
  />
</fieldset>
`;

exports[`TextBox should render with helper text below 1`] = `
<fieldset
  className="css-ur5llr"
>
  <label
    className="css-1a6u95k"
  />
  <input
    className="css-1cn7waq"
    placeholder="Input short info about you"
    type="text"
    value=""
  />
  <span
    className="css-1eztcs5"
  >
    This is just short info
  </span>
</fieldset>
`;
